qiskit/pulse/parser.py

```diff
@@ -17,141 +17,144 @@
 """Helper function to parse string expression given by backends."""
 
 import ast
+import copy
 import operator
 
 import cmath
 
-from qiskit.exceptions import QiskitError
-
-
-# valid functions
-_math_ops = {
-    'acos': cmath.acos,
-    'acosh': cmath.acosh,
-    'asin': cmath.asin,
-    'asinh': cmath.asinh,
-    'atan': cmath.atan,
-    'atanh': cmath.atanh,
-    'cos': cmath.cos,
-    'cosh': cmath.cosh,
-    'exp': cmath.exp,
-    'log': cmath.log,
-    'log10': cmath.log10,
-    'sin': cmath.sin,
-    'sinh': cmath.sinh,
-    'sqrt': cmath.sqrt,
-    'tan': cmath.tan,
-    'tanh': cmath.tanh,
-    'pi': cmath.pi,
-    'e': cmath.e
-}
-
-# valid binary operations
-_binary_ops = {
-    ast.Add: operator.add,
-    ast.Sub: operator.sub,
-    ast.Mult: operator.mul,
-    ast.Div: operator.truediv,
-    ast.Pow: operator.pow
-}
-
-# valid unary operations
-_unary_ops = {
-    ast.UAdd: operator.pos,
-    ast.USub: operator.neg
-}
-
-
-class QiskitExpression(ast.NodeTransformer):
+from qiskit.pulse.exceptions import PulseError
+
+
+class PulseExpression(ast.NodeTransformer):
```

Comment:
It seems now that your view is that this will be pulse only? Do you see this becoming a general qiskit expression in the future?

Reply:
I think this can be a general parser for string input, but circuit parser receives more pythonic object as @kdk pointed, and I don't think to integrate circuit parser with this. If any string expression exists in circuit, this parser should be used.

Reply:
Ok we can always move this in the future.